### PR TITLE
limits read access into Packet data to Packet.meta.size

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Bytes past `Packet.meta.size` are not valid to read from

#### Summary of Changes
The commit makes the field private and instead provides two methods:
* `Packet::data()` which returns an immutable reference to the underlying
  buffer up to `Packet.meta.size`. The rest of the buffer is not valid to
  read from.
* `Packet::buffer_mut()` which returns a mutable reference to the entirety
  of the underlying buffer to write into. The caller is responsible to
  update `Packet.meta.size` after writing to the buffer.

### CODE REVIEW

**Comment by behzadnouri** [comment]
@jstarry To avoid merge conflicts on your side, I can hold off merging this until your changes are in, if that is something you prefer.

**Comment by jstarry** [comment]
> @jstarry To avoid merge conflicts on your side, I can hold off merging this until your changes are in, if that is something you prefer.

Yeah that would be nice, I'll get mine merged in the next day or two

**Review by t-nelson — APPROVED** [review]

File: streamer/src/recvmmsg.rs
Comment (jstarry) [inline streamer/src/recvmmsg.rs:25]: I think clippy will only complain about this on macos targets, can you drop the `&mut` here?

**Review by jstarry — APPROVED** [review]
lgtm after the clippy fix

File: streamer/src/recvmmsg.rs
Comment (behzadnouri) [reply streamer/src/recvmmsg.rs]: done

**Review by steviez — APPROVED** [review]
LGTM

**Pull request merged**
